django/cantusdb_project/main_app/views/chant.py

```diff
@@ -1696,8 +1699,12 @@ def form_valid(self, form):
             return super().form_invalid(form)
 
     def get_success_url(self):
-        # stay on the same page after save
-        return self.request.get_full_path()
+        next_url = self.request.GET.get("ref")
```

Comment:
I don't think I follow the logic here...nothing wrong with the code, but can you explain?

Reply:
Editing a chant from the Detail page does take you back to that Detail page after saving, I thought...

Reply:
huh! This makes sense (I think my understanding of how things work is shaped by my experience using the Edit-a-bunch-of-chants-in-a-single-source page exclusively during my time as an RA for CantusDB).

This should be an easy fix. Let me open an issue.